pyscriptjs/Makefile

```diff
@@ -0,0 +1,84 @@
+tag := latest
+git_hash ?= $(shell git log -1 --pretty=format:%h)
+
+base_dir ?= $(shell git rev-parse --show-toplevel)
+src_dir ?= $(base_dir)/src
+examples ?=  $(base_dir)/examples
+app_dir ?= $(shell git rev-parse --show-prefix)
+
+CONDA_ENV ?= ./env
+env := $(CONDA_ENV)
+
+
+.PHONY: clean
+clean:
+	find . -name \*.py[cod] -delete
+	rm -rf .pytest_cache .coverage coverage.xml
+
+clean-all: clean
+	rm -rf $(env) *.egg-info
+
+.PHONY: setup
+setup:
+	@if [ -z "$${CONDA_SHLVL:+x}" ]; then echo "Conda is not installed." && exit 1; fi
+	$(CONDA_EXE) env $(shell [ -d $(env) ] && echo update || echo create) -p $(env) --file environmentwh.yml
+
+.PHONY: shell
+shell:
+	@export CONDA_ENV_PROMPT='<{name}>'
+	@echo 'conda activate $(env)'
+
+.PHONY: dev
+dev:
+	npm run dev
+
+.PHONY: build
+build:
+	npm run build
+
+.PHONY: test
+test:
+	@echo "Tests are coming :( this is a placeholder and it's meant to fail!"
+	$(env)/bin/pytest -vv $(ARGS) tests/ --log-cli-level=warning
+
+.PHONY: test-py
+test-py:
+	@echo "Tests are coming :( this is a placeholder and it's meant to fail!"
+	$(env)/bin/pytest -vv $(ARGS) tests/ --log-cli-level=warning
+
+.PHONY: test-ts
+test-ts:
+	@echo "Tests are coming :( this is a placeholder and it's meant to fail!"
+	npm run tests
+
+.PHONY: fmt
```

Comment:
My preference would be to delegate the formatting and linting to `pre-commit`, so it will happen automatically.

I plan to configure that. If you'd still like those in the `Makefile`, I'd recommend we modify these to be e.g. `pre-commit run black --all`.

Based on your preference, I can modify the `Makefile` when I do that work.

Reply:
+1 on the `pre-commit` part of it. In general I like to be able to lint on demand so, if others don't see any harm with it, I'd keep both options.

Reply:
Works to keep it, I can update when I implement the `pre-commit` settings.